docs/source/integrations/triton.rst

```diff
@@ -1,3 +1,599 @@
+
+The argument ``model_repository`` is the path to said model repository that Triton can use to serve the model. Note that ``model_repository`` also
+supports S3 path:
+
+
+.. code-block:: python
+
+   import bentoml
+
+   triton_runner = bentoml.triton.Runner("triton_runner", model_repository="s3://bucket/path/to/model_repository")
+
+.. note::
+
+   Currently, the Triton integration requires one to setup the model repository manually.
+
+From a developer perspective, remote invocation of Triton runners is similar to invoking any other BentoML runners. This means
+one can treat Triton as a standalone runtime.
+
+.. note::
+
+   By default, ``bentoml.triton.Runner`` will run the ``tritonserver`` with gRPC protocol. To use HTTP/REST protocol, provide ``tritonserver_type=''http'`` to the ``Runner`` constructor.
+
+   .. code-block:: python
+
+      import bentoml
+
+      triton_runner = bentoml.triton.Runner("triton_runner", model_repository="/path/to/model_repository", tritonserver_type="http")
+
+
+Triton Runner signatures
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Normally in a BentoML Runner, one can access the model signatures directly from the runners attributes. For example, the model signature ``predict``
+of a ``iris_classifier_runner`` (as often seen in our :ref:`tutorial <tutorial:Creating a Service>`) can be accessed as ``iris_classifier_runner.predict.run``.
+
+However, Triton runner's attributes represent individual models defined under the model repository. For example, if the model repository has the following structure:
+
+.. code-block:: prolog
+
+    model_repository
+    ├── onnx_mnist
+    │   ├── 1
+    │   │   └── model.onnx
+    │   └── config.pbtxt
+    ├── tensorflow_mnist
+    │   ├── 1
+    │   │   └── model.savedmodel/
+    │   └── config.pbtxt
+    └── torchscript_mnist
+        ├── 1
+        │   └── model.pt
+        └── config.pbtxt
+
+Then each model inference can be accessed as ``triton_runner.onnx_mnist.<run|async_run>``, ``triton_runner.tensorflow_mnist.<run|async_run>``, and ``triton_runner.torchscript_mnist.<run|async_run>``.
+
+An example to demonstrate how to call the Triton runner:
+
+.. code-block:: python
+
+    import bentoml
+    import numpy as np
+
+    @svc.api(
+        input=bentoml.io.Image.from_sample("./data/0.png"), output=bentoml.io.NumpyNdarray()
+    )
+    async def bentoml_torchscript_mnist_infer(im: Image) -> NDArray[t.Any]:
+        arr = np.array(im) / 255.0
+        arr = np.expand_dims(arr, (0, 1)).astype("float32")
+        InferResult = await triton_runner.torchscript_mnist.async_run(arr)
+        return InferResult.as_numpy("OUTPUT__0")
+
+There are a few things to note here:
+
+1. Triton runners should only be called **lazily**. In other words, if ``triton_runner.torchscript_mnist.async_run`` is invoked in the
+   global scope, it will not work.
+
+   As a result, ``triton_runner.init_local()`` will not work in development mode.
```

Comment:
Maybe it is more clear to say that the Triton runner cannot be initialized locally since it is not natively implemented in Python.